Description: gconf -> mateconf, gnome-session -> mate-session
 A small patch to ensure compatability with the MATE desktop environment
Author: Jasmine Hassan <jasmine.aura@gmail.com>
Last-Update: 2010-10-11
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/config/config
+++ b/config/config
@@ -8,8 +8,8 @@ backend = kconfig
 integration = true
 plugin_list_autosort = true
 
-[gnome_session]
-backend = gconf
+[mate_session]
+backend = mateconf
 integration = true
 plugin_list_autosort = true
 
--- a/include/ccs-backend.h
+++ b/include/ccs-backend.h
@@ -65,7 +65,7 @@ struct _CCSBackendVTable
     Bool profileSupport;
 
     /* something like a event loop call for the backend,
-       so it can check for file changes (gconf changes in the gconf backend)
+       so it can check for file changes (mateconf changes in the mateconf backend)
        no need for reload settings signals anymore */
     CCSExecuteEventsFunc executeEvents;
 
--- a/src/compiz.cpp
+++ b/src/compiz.cpp
@@ -832,7 +832,7 @@ addPluginFromPB (CCSContext * context,
     if (ccsFindPlugin (context, name))
 	return;
 
-    if (!strcmp (name, "ini") || !strcmp (name, "gconf") ||
+    if (!strcmp (name, "ini") || !strcmp (name, "mateconf") ||
 	!strcmp (name, "ccp") || !strcmp (name, "kconfig"))
 	return;
 
@@ -2486,7 +2486,7 @@ addPluginFromXMLNode (CCSContext * conte
 	return FALSE;
     }
 
-    if (!strcmp (name, "ini") || !strcmp (name, "gconf") ||
+    if (!strcmp (name, "ini") || !strcmp (name, "mateconf") ||
 	!strcmp (name, "ccp") || !strcmp (name, "kconfig"))
     {
 	free (name);
@@ -2894,7 +2894,7 @@ addPluginNamed (CCSContext * context, ch
     if (ccsFindPlugin (context, name))
 	return;
 
-    if (!strcmp (name, "ini") || !strcmp (name, "gconf") ||
+    if (!strcmp (name, "ini") || !strcmp (name, "mateconf") ||
 	!strcmp (name, "ccp") || !strcmp (name, "kconfig"))
 	return;
 
--- a/src/config.c
+++ b/src/config.c
@@ -65,9 +65,9 @@ getSectionName (void)
 	return section;
     }
 
-    profile = getenv ("GNOME_DESKTOP_SESSION_ID");
+    profile = getenv ("MATE_DESKTOP_SESSION_ID");
     if (profile && strlen (profile))
-	return strdup ("gnome_session");
+	return strdup ("mate_session");
 
     profile = getenv ("KDE_SESSION_VERSION");
     if (profile && strlen (profile) && strcasecmp (profile, "4") == 0)
